src/nvim/strings.c

```diff
@@ -45,6 +45,41 @@
 #include "nvim/os/os.h"
 #include "nvim/os/shell.h"
 
+/// Compares two strings for equality.
+bool strequal(const char *a, const char *b)
+  FUNC_ATTR_NONNULL_ALL FUNC_ATTR_PURE
+{
+  return strcmp(a, b) == 0;
+}
+
+/// Compares two sized strings for equality.
+bool strnequal(const char *a, const char *b, size_t size)
+  FUNC_ATTR_NONNULL_ALL FUNC_ATTR_PURE
+{
+  return strncmp(a, b, size) == 0;
+}
+
+/// Compares two strings for equality ignoring differences in case.
+bool striequal(const char *a, const char *b)
+  FUNC_ATTR_NONNULL_ALL FUNC_ATTR_PURE
+{
+  return strcasecmp(a, b) == 0;
```

Comment:
I cannot find them being supported even in MSVC 2015 (possibly @equalsraf could give the final OK, but I hear his dev environment is a bit messy atm). So the function check macro's do still make sense. Possibly in such a way:

```c
#if defined(HAVE_STRCASECMP)
  #define XSTRCASECMP strcasecmp
#elif defined(HAVE_STRICMP)
  #define XSTRCASECMP _stricmp
#else
  #error "Your compiler or runtime environment does not support strcasecmp()/stricmp()."
#endif
```

Though I also think it looks really ugly.